daemon/daemon.go

```diff
@@ -168,21 +175,22 @@ func (d *Daemon) queueJob(do DaemonJobFunc) job.ID {
 
 // Apply the desired changes to the config files
 func (d *Daemon) UpdateManifests(spec update.Spec) (job.ID, error) {
+	ctx, cancel := newDaemonContext()
```

Comment:
Needs a `defer cancel()` or we'll leak contexts.

Reply:
The cancel is passed into the `d.queueJob(ctx, cancel, ...` below.
I had to do it this way because the job runs asynchronously. If you do a `defer cancel()` then it would cancel the context before it has been used by the job runner. (I made this mistake! :-) )

Reply:
Ah, got it :)

Reply:
This is going to give jobs a deadline from the point the job is queued, including time spent in the queue. Is that what you wanted?